docs/howto_bentoml.md

```diff
@@ -0,0 +1,295 @@
+Outputs must be formated as a JSON object with a key "predictions" containing a list of values.
+
+Example of input:
+
+```json title="query.json"
+{
+  "instances": [
+    [5.1, 3.5, 1.4, 0.2],
+    [4.9, 3.0, 1.4, 0.2]
+  ]
+}
+```
+
+Example of response:
+
+```json
+{
+  "predictions": [
+    "setosa",
+    "setosa"
+  ]
+}
+```
+
+So we need to write a service that will take the input, format it to a list of lists, and then call the BentoML bundle.
+Then, we need to format the output to the VertexAI format.
+
+You can use pydantic to validate the input and output of your API.
+This will also enrich the API Swagger documentation that is automatically generated.
+
+```py title="service.py"
+import bentoml
+from bentoml.io import JSON
+from pydantic import BaseModel
+from typing import List
+
+# write data model in a separate file for better code readability
+class Query(BaseModel):
+    instances: List[List[float]]
+
+class Response(BaseModel):
+    predictions: List[str]
+
+# Load the BentoML bundle
+iris_clf_runner = bentoml.sklearn.get("iris_clf:latest").to_runner()
+input_schema = JSON(pydantic_model=Query)
+output_schema = JSON(pydantic_model=Response)
+
+svc = bentoml.Service("iris_classifier", runners=[iris_clf_runner])
+
+@svc.api(input=input_schema, output=output_schema)
+def classify(input_series: dict) -> dict:
+    input_series = input_series["instances"]
+    return {"predictions": iris_clf_runner.predict.run(input_series)}
+```
+
+You can test the service locally using the following command: `bentoml serve service.py:svc --reload`
+
+### 3. Write the bentofile.yaml file
+
+```yaml title="bentofile.yaml"
+service: "service.py:svc"
+labels:
+  owner: bentoml-team
+  project: gallery
+include:
+  - "*.py"
+python:
+  packages:
+    - scikit-learn
+    - pandas
+```
+
+### 4. Build the Docker image
+
+Here are the steps to build the Docker image and run it locally:
+
+```bash
+VERSION="0.0.1"
+GCP_PROJECT="my-gcp-project"
+SERVICE="iris_classifier"
+YAML_PATH="bentofile.yaml"
+IMAGE_URI=eu.gcr.io/$GCP_PROJECT/$SERVICE:$VERSION
+
+bentoml build -f $YAML_PATH ./src/ --version $VERSION
+bentoml serve $SERVICE:latest --production
+bentoml containerize $SERVICE:$VERSION -t $IMAGE_URI
+```
+
+However, it's not handy to use the CLI if you want to do this step in a Python Script, and there is a lack of documentation on that.
+So I wrote a [Python script](docs/assets/howto_bentoml/workflows/build_bento.py) to do this step, that can be used in a VertexAI component.
+
+Under the hood, when you do the `bentoml containerize` command, docker actually builds the image.
+If you want to use this as a VertexAI component, you cannot rely on Docker as you are already in a container. 
+The workaround here is to use Cloud Build to build the image.
+
+Check it out: [docs/assets/howto_bentoml/workflows/build_bento.py](docs/assets/howto_bentoml/workflows/build_bento.py)
```

Comment:
Issue with this one to. I see it as a link but not redirecting to the file (but maybe because i'm using it locally, to be further checked)